### ISSUE TITLE
Update LocalFileStorage to work with django 3.2.11

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#18570]
https://www.djangoproject.com/weblog/2022/jan/04/security-releases/ includes a fix (https://github.com/django/django/commit/6d343d01c57eb03ca1c6826318b652709e58a76e) for directory traversal.  

Our LocalFileStorage does a few things that conflict with it:
- uses bytestrings rather the str for paths
- allows us to access paths other than under `MEDIA_ROOT`

**Comment by ioanarusiczki** [comment]
@eviljeff Please add some str for qa. 
Thanks!

**Comment by eviljeff** [comment]
> @eviljeff Please add some str for qa. Thanks!

it's a pretty wide ranging change - anywhere that reads or writes to user provided, or generated, content could be affected.  So check uploads and downloads - in particular downloads of _existing_ content (i.e. before this change landed).
- source code (devhub)
- user picture (user profile, via frontend)
- xpi (devhub; reviewer tools; frontend after it's been approved)
- theme preview generation
- hero image (admin tools; frontend)
- blocklist mlbf generation

We have pretty test good coverage so I'm not super concerned about new uploads/generation of content, but existing content is harder to test against (if we inadvertently changed the save location of a file then the upload and download of a new file would probably work equally well, but the existing file would now be in the "wrong" location)

Marking this as QA extra because it's a bit sprawling.

**Comment by ioanarusiczki** [comment]
@eviljeff 

- source code -> I found some samples uploaded last week while testing for https://github.com/mozilla/addons-server/issues/18489#issuecomment-1005496393. Dowloading worked, also uploading them at submission or from dev hub is working.
- user picture does not seem to be affected, changing , deleting or viewing older images uploaded by other users. 
- no issues at downloading xpi files.
- I did not see issues on theme previews -> they load on all frontend pages, also previews generated at submissions looked alright. 
- the hero images are still displayed in admin tools and frontend, I could upload new samples in admin tools and attach them to promoted addons.
- the [stash](https://settings-cdn.stage.mozaws.net/v1/buckets/blocklists/collections/addons-bloomfilters/records) was generated for new addon blocks, deleted blocks or edited blocks as expected.
